### ISSUE TITLE
Exception thrown by a static @RegisterExtension field is swallowed

### ISSUE THREAD

**Issue by mbaechler** [junit-team/junit5#1666]
## Steps to reproduce

Running this test case

```java
package org.example;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.Extension;
import org.junit.jupiter.api.extension.RegisterExtension;

class ExplosiveTests {

	@RegisterExtension
	static Extension field = new Extension() {
		{
			if (true) {
				throw new Error("boom");
			}
		}
	};

	@Test
	void test() {
	}
}
```

should report: 

```
java.lang.Error: boom
	at org.example.ExplosiveTests$1.<init>(ExplosiveTestClass.java:31)
	at org.example.ExplosiveTests.<clinit>(ExplosiveTestClass.java:28)
```

instead of:

```
java.lang.NoClassDefFoundError: Could not initialize class org.example.ExplosiveTests
```

## Context

 - Used versions (Jupiter/Vintage/Platform): JUnit Jupiter 5.2.0
 - Build Tool/IDE: Intellij Idea 2018.2

## Related Issues

- #602

**Comment by sbrannen** [comment]
FYI: I authored `ExplosiveTests` and verified the behavior against the status quo on `master`.

**Comment by sbrannen** [comment]
See also: #602

**Comment by sbrannen** [comment]
Tentatively slated for 5.4 M1 for _team discussion_.

**Comment by sbrannen** [comment]
Please note that the described (undesirable) behavior only exists if the field is `static`!

If the `@RegisterExtension` field is non-static, JUnit properly propagates the `java.lang.Error: boom` exception.

**Comment by sbrannen** [comment]
The reason for the latter is that `org.junit.platform.commons.util.ReflectionUtils.newInstance(Constructor<T>, Object...)` does _not_ swallow the exception; whereas, `org.junit.platform.commons.util.ReflectionUtils.readFieldValue(Field, T)` silently swallows the exception and returns `Optional.empty()`.

**Comment by sbrannen** [comment]
Fixed for 5.3.2 in commit a64e0b69ee7bc475bfef26d12819378ff292bcd3.
